black_it/loss_functions/base.py

```diff
@@ -66,12 +76,39 @@ def compute_loss(
             )
             weights = self.coordinate_weights
 
+        if self.coordinate_filters is None:
+            # a list of identity functions
+            filters = [None] * num_coords
+        else:
+            nb_coordinate_filters = len(self.coordinate_filters)
+            assert_(
+                nb_coordinate_filters == num_coords,
+                (
+                    "the length of coordinate_filters should be equal "
+                    f"to the number of coordinates, got {nb_coordinate_filters} and {num_coords}"
+                ),
+                exc_cls=ValueError,
+            )
+            filters = self.coordinate_filters
+
         loss = 0
         for i in range(num_coords):
-            loss += (
-                self.compute_loss_1d(sim_data_ensemble[:, :, i], real_data[:, i])
-                * weights[i]
-            )
+            if filters[i] is None:
```

Comment:
An idea to simplify the code is to move this check in the constructor, pre-process the `coordinate_filters` argument so to have either user-defined filters or the identity functions.

Reply:
Once again, this is not possible since the constructor does not know the number of coordinates unfortunately

Reply:
I think this part is possible because in case `filters[i]` is None, we can assign a lambda function like `filters[i] = lambda x: x`, and then just use it rather than checking `if filters[i] is None`